Feature/300 wiennerisches diarium historical toponyms

Sarah1108 opened the pull request:
Data Preprocessing cleaned
Minio missing

anasedova commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/Sterbelisten_Preprocessing.ipynb:
```diff
@@ -0,0 +1,395 @@
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
```
remove the cell if not needed

Sarah1108 replied:
done!

anasedova commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/Sterbelisten_Preprocessing.ipynb:
```diff
@@ -0,0 +1,395 @@
+    "with jsl.open('timemachine_evaluation_v1_edited_corrected.jsonl','r') as reader: #read in jsonl with jsonlines\n",
+    "    df = pd.json_normalize(reader) \n",
+    "\n",
+    "### I don't understand where the tagged toponym is or whether there is any"
```
we can discuss it on Monday as well

Sarah1108 replied:
Yes, I added an explanation for why this data is not suitable as test data for our task

anasedova reviewed (changes requested):
A general note: have a look at some tutorials that are already in Knodle (e.g. [IMDB tutorial](https://github.com/knodle/knodle/blob/develop/examples/data_preprocessing/imdb_dataset/Prepare_weak_IMDB_data.ipynb)), or the other DS tutorials in PRs that were already reviewed a couple of times #299, #295 etc. The main idea of a tutorial is to tell a story about data preprocessing - how it was realized, why. It should be interesting to read it as a narrative. Apart from that, the tutorial will be used by people who have never worked with this type of data, or with Knodle, or with weak supervision as well - everything should be clear for them. Examples are always good as well. 

Also: all .html and .json files should be removed from GitHub, stored in minio, and downloaded from there at the beginning of the tutorial. In general, the best practice is to store only scripts in git.

Sarah1108 commented on examples/trainer/[WIP]_baseline/bert/bert.ipynb:
```diff
@@ -484,7 +484,7 @@
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.7.9"
+   "version": "3.8.3"
```
It is not needed and happend whilst first trying knodle!

anasedova replied:
then please rollback them

Sarah1108 commented on examples/data_preprocessing/imdb_dataset/Prepare_weak_IMDB_data.ipynb:
```diff
@@ -34,7 +34,7 @@
   {
    "cell_type": "code",
-   "execution_count": 2,
+   "execution_count": 1,
```
same here, sorry, not needed!

anasedova replied:
then please rollback them

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/Sterbelisten_Preprocessing.ipynb:
```diff
@@ -0,0 +1,395 @@
+   "source": [
+    "'''\n",
+    "X Matrix\n",
+    "Dimensions sentesesxsentenses (one column sample df, len(sample))\n",
```
I added the change to the note book

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/Sterbelisten_Preprocessing.ipynb:
```diff
@@ -0,0 +1,395 @@
+   "outputs": [],
+   "source": [
+    "'''\n",
+    "Here I am building the Z Matrixes by looping through all samples.\n",
```
I added this to the note book!

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/Sterbelisten_Preprocessing.ipynb:
```diff
@@ -0,0 +1,395 @@
+    "file = codecs.open('annotations_3-21_v4.html', \"r\", \"utf-8\")\n",
+    "sterbelisten_html = file.read()\n",
+    "\n",
+    "'''Cleaning the text first from both html taggs and xml tags and than in a second step clean the intro and replace with /n'''\n",
```
I moved it to a new cell

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/Sterbelisten_Preprocessing.ipynb:
```diff
@@ -0,0 +1,395 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "'''Load html file'''\n",
```
I replaced the comments with #

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/Sterbelisten_Preprocessing.ipynb:
```diff
@@ -0,0 +1,395 @@
+    "**5. Building test data** \n",
+    "\n",
+    "manually checked! \n",
+    "There is the 'timemachine_evaluatuin_v1_edited_corrected.jsonl which should contain manually corrected place names but I don't understand where the actual word is tagged or how to filter out that information."
```
I hope this is explained enough now, I added it to the notebook in detail in the section 5

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/Sterbelisten_Preprocessing.ipynb:
```diff
@@ -0,0 +1,395 @@
+    "\n",
+    "**2. Read in already extracted placenames**\n",
+    "Extract all \\<mark> words from HTML**\n",
+    "Using their code\n",
```
sorry I cleand that whole section to make it more structured

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/Sterbelisten_Preprocessing.ipynb:
```diff
@@ -0,0 +1,395 @@
+    "\n",
+    "**1. Read in the html file**\n",
+    "Strip of tags and clean the text with re\n",
+    "Expand abbreviations- if possible?\n",
```
I explained, that we don't want to expand, as this would mean normalisation

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/Sterbelisten_Preprocessing.ipynb:
```diff
@@ -0,0 +1,395 @@
+    "##### What to do\n",
+    "\n",
+    "**1. Read in the html file**\n",
+    "Strip of tags and clean the text with re\n",
```
I replaced that

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/Sterbelisten_Preprocessing.ipynb:
```diff
@@ -0,0 +1,395 @@
+    "There are .csv files with Toponyms extracted from Wien Geschichte Wiki.\n",
+    "\n",
+    "##### What we want\n",
+    "A trainig and test set containing three matrixes:\n",
```
I changed all those parts to bullet lists

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/Sterbelisten_Preprocessing.ipynb:
```diff
@@ -0,0 +1,395 @@
+    "There is a list with rules which are combines Regex and other expressions. \n",
+    "There is an html file that has \\<mark> on most toponyms but sometimes on persons and other names too.\n",
+    "There are defined classes (Toponym/no Toponym).\n",
+    "There are .csv files with Toponyms extracted from Wien Geschichte Wiki.\n",
```
I started calling it research group by the ACDH-CH

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/Sterbelisten_Preprocessing.ipynb:
```diff
@@ -0,0 +1,395 @@
+    "\n",
+    "##### What we have \n",
+    "There is a list with sucessfully extracted toponyms, which are already normalised orthography.\n",
+    "There is a list with rules which are combines Regex and other expressions. \n",
```
I restructured these parts, hopefully it is more understandable now

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/Sterbelisten_Preprocessing.ipynb:
```diff
@@ -0,0 +1,395 @@
+    "\n",
+    "\n",
+    "##### What we have \n",
+    "There is a list with sucessfully extracted toponyms, which are already normalised orthography.\n",
```
I went into detail here

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/Sterbelisten_Preprocessing.ipynb:
```diff
@@ -0,0 +1,395 @@
+    "\n",
+    "### Task\n",
+    "\n",
+    "Filter out Toponyms from the Sterbelisten. For this sequence labeling task we need to first tokenize the words and use the rules (patterns and keywords) made by the team (Nina) to fit to our classes (Toponym or not).\n",
```
I added an example of the html raw data we have an how we clean it stwp by step

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/Sterbelisten_Preprocessing.ipynb:
```diff
@@ -0,0 +1,395 @@
+    "\n",
+    "The digitalised Wiener Zeitung is partially available as TEI XML as well as an extracted HTML on Github. Starting from 1703, there is representative data.This data set contains only a few newspapers per year starting from 1706.\n",
+    "\n",
+    "Github Repository: https://github.com/acdh-oeaw/sterbelisten\n",
```
I added that

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/Sterbelisten_Preprocessing.ipynb:
```diff
@@ -0,0 +1,395 @@
+    "## Wiener Diarum Sterbelisten\n",
+    "-----------------------------------------------------------------------------------------------\n",
+    "\n",
+    "### Data\n",
```
I wrote something regarding the data I hope it is not too much but once started it is difficult to keep it too short

anasedova commented on examples/trainer/simple_auto_trainer/auto_trainer_tutorial.ipynb:
```diff
@@ -2,7 +2,6 @@
  "cells": [
   {
    "cell_type": "markdown",
```
if you don't need them (and I suppose, you don't :)) - please rollback them

anasedova commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb:
```diff
@@ -0,0 +1,512 @@
+    }
+   ],
+   "source": [
+    "def get_location_id(location: List, locations: List) -> int:\n",
```
but later you create locations = {} - is it Dict or List?

anasedova commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb:
```diff
@@ -0,0 +1,512 @@
+    "    for word in location:\n",
+    "        if word not in locations:\n",
+    "            locations[word]= len(locations)\n",
+    "            return locations[word]\n",
```
you should also add `else` branch, i.e. what will the function return if `word in locations`?

anasedova replied:
also: I would suggest changing the naming a bit because `location` and `locations` sound too similar

anasedova replied:
also-2: it is better to convert this part of code into the separate function + write a test for it (let's talk about it)

anasedova replied:
docstring: 
```
"""
[discription what function does]
Args:
    curr_location: 
    all_locations: 
Returns: 
"""
```

anasedova commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb:
```diff
@@ -0,0 +1,512 @@
+    "b= np.full(len(locations),1)\n",
+    "\n",
+    "T = np.stack((b,a.T))\n",
+    "T.T.shape\n"
```
bad naming, better to change `T` to  e.g. `t_matrix` (and then `Z`to `z_matrix` for consistency reason)

anasedova commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb:
```diff
@@ -0,0 +1,512 @@
+    "locations_list = list(range(0,(len(locations))))\n",
+    "\n",
+    "\n",
+    "for sample in samples:\n",
```
again: it is better to convert this part of code into the separate function + write a test for it (let's talk about it)

anasedova commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_wiener_diarum_toponym.py:
```diff
@@ -0,0 +1,158 @@
+    Z = np.array(matched_loc)
+    Z = np.reshape(Z, (i, len(locations_list)))
+
+# Needed: a way to save them with their index number!
```
is it a relevant comment?

anasedova reviewed (changes requested):
Great, look already much better! 
Here are some of my comments, let's talk about it (especially test creation)
Also: I have left the comments for jupyter notebook only, but many of them are also valid for .py script - please make the appropriate corrections there as well.

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb:
```diff
@@ -0,0 +1,512 @@
+    "\n",
+    "### Task\n",
+    "\n",
+    "In this Tutorial we want to prepare the data provided by the research group of the ACDH-CH in order to apply sequence labeling to it and fit the data into the knodle frame work.\n",
```
I replaced this

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb:
```diff
@@ -0,0 +1,512 @@
+    "\n",
+    "In this Tutorial we want to prepare the data provided by the research group of the ACDH-CH in order to apply sequence labeling to it and fit the data into the knodle frame work.\n",
+    "\n",
+    "Our goal is to automatically fiter out place names from the obituaries. These lists are in every newspaper issue and contain several notes about the death of people. Usually they contain the name, age, reason and place of death. By filtering out the place names automatically, one can extract historical knowledge about the city of Vienna and its deveopement as well as orthographical changes a ...
```
I replaced this too

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb:
```diff
@@ -0,0 +1,512 @@
+    "- X Matrix containing the samples (pandas data frame).\n",
+    "\n",
+    "\n",
+    "##### What to do\n",
```
and this

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb:
```diff
@@ -0,0 +1,512 @@
+    "\n",
+    "As the ACDH-CH research group is activley working on their project, the Github repository is constantly changing and developing. For our Tutorial we will only use the .html file as it contains the marked place names we need to build our training data.\n",
+    "\n",
+    "##### What we want\n",
```
and this too

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb:
```diff
@@ -0,0 +1,512 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Cleaning the text first from both html tags and xml tags and then in a second step clean the intro and replace with /n\n",
```
checked the " " after each #

Sarah1108 commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb:
```diff
@@ -0,0 +1,512 @@
+   "source": [
+    "print(f\"Number of <mark> tags: {len(re.findall('<mark>',sterbelisten_html))}\")\n",
+    "print(f\"Number of </mark> tags: {len(re.findall('</mark>',sterbelisten_html))}\")\n",
+    "print((f\"Number of unclosed tags: {len(re.findall('<mark>',sterbelisten_html))-len(re.findall('</mark>',sterbelisten_html))}\"))"
```
I added the cleaning of the data set from the unclosed tags here

Sarah1108 commented:
I seperated the sentence preprocessing, the building of the dic with all place names and creating the labels for each sentence into three functions, I collect all imports at the beginning of the tutorial with a description of all files in the folder, I added a line to save the list with all z_matrices as .lib file and added a brief description of all files available on minio. 

I improved the code and got rid of any not needed loop through all sentences and "cleaned up the code", as we discussed on friday.

I fixed importing the html file and downloading it from minio.

I tried to write a test for the main function of getting the labels, but it is not working yet. 

I put together a requirements file and updated the .py file.

I updated the minio folder.

vasilikikou commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb:
```diff
@@ -0,0 +1,1246 @@
+    }
+   ],
+   "source": [
+    "# define the path to the folder where the data will be stored\n",
```
From the Pep 8 style Guide for comments: "Comments should be complete sentences. The first word should be capitalized, unless it is an identifier that begins with a lower case letter."

vasilikikou commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb:
```diff
@@ -0,0 +1,1246 @@
+    "for file in tqdm(files):\n",
+    "    client.fget_object(\n",
+    "        bucket_name=\"knodle\",\n",
+    "        object_name=os.path.join(\"datasets/wiener_diarum_toponyms/\",file),\n",
```
```suggestion
    "        object_name=os.path.join(\"datasets/wiener_diarum_toponyms/\", file),\n",
```

vasilikikou commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb:
```diff
@@ -0,0 +1,1246 @@
+   "source": [
+    "# Load html file\n",
+    "\n",
+    "with open(os.path.join(data_path, file[-1]), 'r', encoding = \"utf-8\" ) as f:\n",
```
Choose single-quotes or double-quotes and stick to that throughout the code.

vasilikikou replied:
Don't use spaces around the "=" sign when used to indicate an argument. Also the space before the closing parenthesis should be removed.

vasilikikou commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb:
```diff
@@ -0,0 +1,1246 @@
+    "# Cleaning the text first from both html tags and xml tags and then in a second step clean the intro and replace with /n\n",
+    "\n",
+    "# get rid of the title of the issue and the xml tags by just replacing the whole intro with \"\"\n",
+    "sterbelisten_strip_html_1 = re.sub(r\"<h2>.*</h2><h3>.*xml \\|.*\\d+</h3><br/>\",\"\",sterbelisten_html)\n",
```
```suggestion
    "sterbelisten_strip_html_1 = re.sub(r\"<h2>.*</h2><h3>.*xml \\|.*\\d+</h3><br/>\", \"\", sterbelisten_html)\n",
```

vasilikikou replied:
Add a space after the comma throughout the code.

vasilikikou commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb:
```diff
@@ -0,0 +1,1246 @@
+    "sterbelisten_strip_html_1 = re.sub(r\"<h2>.*</h2><h3>.*xml \\|.*\\d+</h3><br/>\",\"\",sterbelisten_html)\n",
+    "\n",
+    "# now working on all the tags around the sentences and weird characters within the words\n",
+    "sterbelisten_strip_html1 = re.sub(r\"<hr/>|<p>|</p>|#+|=| =|<html>|</html>|\\r|\\b \\.|\\b# \\.|(|)\",\"\",sterbelisten_strip_html_1)#<mark>|</mark>\n",
```
```suggestion
    "sterbelisten_strip_html1 = re.sub(r\"<hr/>|<p>|</p>|#+|=| =|<html>|</html>|\\r|\\b \\.|\\b# \\.|(|)\", \"\", sterbelisten_strip_html_1)#<mark>|</mark>\n",
```

vasilikikou commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb:
```diff
@@ -0,0 +1,1246 @@
+    "# now working on all the tags around the sentences and weird characters within the words\n",
+    "sterbelisten_strip_html1 = re.sub(r\"<hr/>|<p>|</p>|#+|=| =|<html>|</html>|\\r|\\b \\.|\\b# \\.|(|)\",\"\",sterbelisten_strip_html_1)#<mark>|</mark>\n",
+    "\n",
+    "sterbelisten_strip_html2 = re.sub(r\"    \\b\",\"\",sterbelisten_strip_html1)"
```
```suggestion
    "sterbelisten_strip_html2 = re.sub(r\"    \\b\", \"\", sterbelisten_strip_html1)"
```

vasilikikou commented on examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb:
```diff
@@ -0,0 +1,1246 @@
+    "    \n",
+    "    '''\n",
+    "    # we create a dictionary locations that contains the place name as a key and the ID as a value\n",
+    "    all_loc={}\n",
```
The assignment operator should be surrounded with a single space.

vasilikikou reviewed (changes requested):
Very good Sarah! I added comments regarding some typos in the text and the style in the code. Please check my comments and make the changes I mention throughout the code (also in the .py file), so that it follows the Pep 8 style guide.
